contracts/ProductNft/ProductNft.sol

```diff
@@ -257,10 +267,40 @@ contract ProductNft is ERC1155, Ownable {
     *@param tokenIdList: the array of token Ids that is used to change the 
     *       deployment status of a token launched using the Valorize Token Launcher
     */
-    function switchProductStatusToDeployed(uint256[] memory tokenIdList) public onlyOwner {
-        for(uint256 i=0; i < tokenIdList.length; i++) {
+    function switchProductStatusToDeployed(uint256[] memory tokenIdList) external onlyOwner {
+        for(uint256 i=0; i < tokenIdList.length;) {
             require(ProductStatusByTokenId[tokenIdList[i]] == ProductStatus.ready, "Your token is not ready yet");
             ProductStatusByTokenId[tokenIdList[i]] = ProductStatus.deployed;
+            unchecked {
+                i++;
+            }
         }
     }
+
+    function supportsInterface(bytes4 interfaceId) public view override(ERC1155, IERC165) returns (bool) {
+        return interfaceId == type(IERC2981).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /**
+    * @dev  Information about the royalty is returned when provided with token id and sale price. 
+    *       Royalty information depends on token id: if token id is smaller than 12 than the artist address is given.
+    *       If token id is bigger than 12 than the funds will be sent to the contract that distributes royalties.
+    * @param _tokenId is the tokenId of an NFT that has been sold on the NFT marketplace
+    * @param _salePrice is the price of the sale of the given token id
+    */
+    function royaltyInfo(
+        uint256 _tokenId,
+        uint256 _salePrice
+    ) external view override returns (
+        address,
+        uint256 royaltyAmount
+    ) {
+        royaltyAmount = (_salePrice / 100) * 10;
+        if (_tokenId <= startRarerTokenIdIndex) {
+            return(addressProductNFTArtist, royaltyAmount);
```

Comment:
Looking at this, the product will need to have the same access control as the royalty distributor.